vital/types/camera_rpc.h

```diff
@@ -0,0 +1,183 @@
+#include <vital/types/camera.h>
+#include <vital/types/vector.h>
+#include <vital/logger/logger.h>
+
+
+namespace kwiver {
+namespace vital {
+
+typedef Eigen::Matrix< double, 4, 20 > rpc_matrix;
+
+/// forward declaration of perspective camera class
+class camera_rpc;
+/// typedef for a camera_rpc shared pointer
+typedef std::shared_ptr< camera_rpc > camera_rpc_sptr;
+
+
+// ------------------------------------------------------------------
+/// An abstract representation of perspective camera
+/**
+ * The base class of camera_rpcs is abstract and provides a
+ * double precision interface.  The templated derived class
+ * can store values in either single or double precision.
+ */
+class VITAL_EXPORT camera_rpc : public camera
+{
+public:
+  /// Destructor
+  virtual ~camera_rpc() = default;
+
+  /// Create a clone of this camera_rpc object
+  virtual camera_sptr clone() const = 0;
+
+  // Accessors
+  virtual rpc_matrix rpc_coeffs() const = 0;
+  virtual vector_3d world_scale() const = 0;
+  virtual vector_3d world_offset() const = 0;
+  virtual vector_2d image_scale() const = 0;
+  virtual vector_2d image_offset() const = 0;
+
+  // Vector of the powers of the positions
+  virtual Eigen::Matrix< double, 20, 1 > power_vector( const vector_3d& pt ) const = 0;
+
+  /// Project a 3D point into a 2D image point
+  virtual vector_2d project( const vector_3d& pt ) const;
+
+protected:
+  camera_rpc();
+
+  kwiver::vital::logger_handle_t m_logger;
+
+};
+
+
+/// A representation of a camera
+/**
+ * Contains camera location, orientation, and intrinsics
+ */
+class VITAL_EXPORT simple_camera_rpc :
+  public camera_rpc
+{
+public:
+  /// Default Constructor - creates identity camera
+  simple_camera_rpc ( ) :
+    world_scale_(1.0, 1.0, 1.0),
+    world_offset_(0.0, 0.0, 0.0),
+    image_scale_(1.0, 1.0),
+    image_offset_(0.0, 0.0)
+  {
+    rpc_coeffs_ = rpc_matrix::Zero();
+    rpc_coeffs_(1, 0) = 1.0;
+    rpc_coeffs_(3, 0) = 1.0;
+    rpc_coeffs_(0, 1) = 1.0;
+    rpc_coeffs_(2, 2) = 1.0;
+  }
+
+/// Constructor - direct from coeffs, scales, and offset
+  /**
+   *  This constructor constructs a camera directly from the RPC parameters
+   */
+  simple_camera_rpc ( vector_3d &world_scale, vector_3d &world_offset,
+                      vector_2d &image_scale, vector_2d &image_offset,
+                      rpc_matrix &rpc_coeffs ) :
+    world_scale_( world_scale ),
+    world_offset_( world_offset ),
+    image_scale_( image_scale ),
+    image_offset_( image_offset ),
+    rpc_coeffs_( rpc_coeffs )
+  { }
+
+  /// Constructor - from base class
+  simple_camera_rpc ( const camera_rpc &base ) :
+    world_scale_( base.world_scale() ),
+    world_offset_( base.world_offset() ),
+    image_scale_( base.image_scale() ),
+    image_offset_( base.image_offset() ),
+    rpc_coeffs_( base.rpc_coeffs() )
+  { }
+
+  /// Create a clone of this camera object
+  virtual camera_sptr clone() const
+  { return camera_sptr( new simple_camera_rpc( *this ) ); }
```

Comment:
it would be better to use `std::make_shared<simple_camera_rpc>( *this )` here.  I know `new` is used all over the place for making cameras, but we should probably use make_shared at least on the new uses (and maybe update the rest later).